CollectionManagerDll/Modules/FileIO/FileCollections/OsdbCollectionHandler.cs

```diff
@@ -127,23 +120,49 @@ public void CompressStream(Stream inputStream, Stream outputStream)
 
                 _binWriter.Write(collection.Name);
                 _binWriter.Write(collection.OnlineId);
+                var customFieldDefLookup = new Dictionary<string, CustomFieldDefinition>();
+                foreach(var def in collection.CustomFieldDefinitions)
+                {
+                    customFieldDefLookup.Add(def.Key, def);
+                    _binWriter.Write(def.Key);
+                    _binWriter.Write((byte)def.Type);
+                    _binWriter.Write(def.DisplayText);
+                }
+                // signal end of custom field definitions
+                _binWriter.Write("");
```

Comment:
All collections should share definitions in osdb - how about storing these once at the beginning?